src/Frontend/Core.hs

```diff
@@ -12,6 +13,61 @@ import Lucid
 import Lucid.Base
 import Network.Wai.Internal (Response(ResponseFile, ResponseBuilder, ResponseStream, ResponseRaw))
 import Network.Wai (Middleware)
+import Text.Digestive.View
+
+----------------------------------------------------------------------
+-- building blocks
+
+-- | Render Form based Views 
+class FormPageView p where
+    -- | Generates a Html view from the given view, form action, and the @p@ page
+    formPageView :: (Monad m) => View (HtmlT m ()) -> ST -> p -> HtmlT m ()
+
+-- | The page after submitting a form should be redirected 
+class RedirectOf p where
```

Comment:
sratch my last comment.  i didnt' read the comment, and thought each page had its own path stored here, which is not the case.